services/bls_aggregation/blsagg.go

```diff
@@ -350,8 +344,28 @@ func (a *BlsAggregatorService) singleTaskAggregatorGoroutineFunc(
 				// happens..
 				continue
 			}
-			// after verifying signature we aggregate its sig and pubkey, and update the signed stake amount
+
+			// check if the operator has already signed for this digest
 			digestAggregatedOperators, ok := aggregatedOperatorsDict[taskResponseDigest]
+			if ok {
+				if digestAggregatedOperators.signersOperatorIdsSet[signedTaskResponseDigest.OperatorId] {
+					a.logger.Warn(
+						"Duplicate signature received",
+						"operatorId", fmt.Sprintf("%x", signedTaskResponseDigest.OperatorId),
+						"taskIndex", taskIndex,
+					)
+					signedTaskResponseDigest.SignatureVerificationErrorC <- fmt.Errorf("duplicate signature from operator %x for task %d", signedTaskResponseDigest.OperatorId, taskIndex)
+					continue
+				}
+			}
+
+			err = a.verifySignature(taskIndex, signedTaskResponseDigest, operatorsAvsStateDict)
+			signedTaskResponseDigest.SignatureVerificationErrorC <- err
```

Comment:
```suggestion
                         // return the err (or nil) to the operator, and then proceed to do aggregation logic asynchronously (when no error)
			signedTaskResponseDigest.SignatureVerificationErrorC <- err
```

Reply:
Can we add a comment like this? Even I was confused reading my old code as to why I was returning a potentially nil error to the sender here on first reading haha

Reply:
https://github.com/Layr-Labs/eigensdk-go/pull/345/commits/fb2628d6616b8d40ea1954f9936003578788badc